bindgen/ir/ty.rs

```diff
@@ -1214,7 +1216,32 @@ impl Type {
                 ty.elem_type()
                     .map_or(false, |element| element.is_const()));
 
-        let ty = Type::new(name, layout, kind, is_const);
+        let mut ty = Type::new(name.clone(), layout.clone(), kind);
+
+        if is_const {
```

Comment:
So we should be able to abstract this into a `maybe_qualify` helper or something. Otherwise it feels very clunky, wdyt?